src/main/java/mate/academy/spring/dao/ShoppingCartDao.java

```diff
@@ -1,10 +1,11 @@
 package mate.academy.spring.dao;
 
+import java.util.Optional;
 import mate.academy.spring.model.ShoppingCart;
 import mate.academy.spring.model.User;
 
 public interface ShoppingCartDao extends GenericDao<ShoppingCart> {
-    ShoppingCart getByUser(User user);
+    Optional<ShoppingCart> getByUser(User user);
```

Comment:
We always have shopping cart for user as we create it during user registration, so lets return just ShoppingCart here

Reply:
Agree